src/server/adapters/logger/logProvider.ts

```diff
@@ -54,14 +64,72 @@ const getLogDNA = ({ apiKey, environment }: LogDNAParams): LogProvider => {
     };
 };
 
+const getPino = ({ logFilePath, environment }: PinoParams): LogProvider => {
+    const nrPinoConf = nrPino();
+
+    const pinoConf = {
+        ...nrPinoConf,
+        formatters: {
+            ...nrPinoConf.formatters,
+            level: (label: string) => ({ level: label })
+        },
+        base: {
+            service: environment === "prod" ? "microsoft-teams-app" : `microsoft-teams-app-${environment}`
+        }
+    };
+    let stream: undefined | SonicBoom;
+    const isLocal = ["local", "docker-local"].includes(environment);
+
+    if (!isLocal) {
+        stream = pino.destination(logFilePath);
+
+        /*
+         This is required for pino to keep on writing logfile after log rotation.
+         'SIGHUP' event be triggered by logrotate after log rotation.
+         */
+        process.on("SIGHUP", () => (stream as SonicBoom).reopen());
+    }
```

Comment:
We can use `logFilePath` variable to decide whether to use stream or not. Also, We don't need to import another library, `sonic-boom`, to get the type. We can rewrite here:
```suggestion
    const stream = logFilePath ? pino.destination(logFilePath) : undefined;
    if (stream) {
        /*
         This is required for pino to keep on writing logfile after log rotation.
         'SIGHUP' event be triggered by logrotate after log rotation.
         */
        process.on("SIGHUP", () => stream.reopen());
    }
```